profile/profile_test.go

```diff
@@ -688,6 +810,123 @@ func locationHash(s *Sample) string {
 	return tb
 }
 
+func TestInferUnits(t *testing.T) {
+	var tagFilterTests = []struct {
+		name             string
+		tagVals          []map[string][]int64
+		tagUnits         []map[string][]string
+		wantUnits        map[string]string
+		wantIgnoredUnits map[string][]string
+	}{
+		{
+			"test1",
```

Comment:
Name the tests by the actual thing they test? "test1", "test2" and so on is not going to be useful in a couple months.